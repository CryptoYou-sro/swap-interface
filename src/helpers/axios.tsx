import { useCallback, useEffect, useState } from 'react';
import { Buffer } from 'buffer';
import {
	BASE_URL,
	BINANCE_DEV_URL,
	BINANCE_SCRIPT,
	BIZ_ENTRY_KEY,
	BINANCE_EXCHANGE_INFO,
	BINANCE_PRICE_TICKER
} from './index';
import axios from 'axios';
import destinationNetworks from '../data/destinationNetworks.json';

export enum STATUS_ENUM {
	NONCE = 'NONCE',
	AUTH = 'AUTH',
	PASS = 'PASS'
}

type LocalStorageProps = {
	isKyced: boolean;
	refresh: string;
	access: string;
	account: string;
};

export const apiCall = {
	getNonce: 'nonce?address=',
	auth: 'auth',
	kycToken: 'kyc/token',
	kycStatus: 'kyc/status',
	refresh: 'refresh'
};

export const getMetamaskMessage = (nonce: string): string =>
	`0x${Buffer.from('Please sign this one time nonce: ' + nonce, 'utf8').toString('hex')}`;

export const loadBinanceKycScript = (cb?: any) => {
	const existingId = document.getElementById('binance-kcy-script');

	if (!existingId) {
		const binanceSdkScript = document.createElement('script');
		binanceSdkScript.src = BINANCE_SCRIPT;
		binanceSdkScript.id = 'binance-kcy-script';
		document.body.appendChild(binanceSdkScript);

		binanceSdkScript.onload = () => {
			if (cb) cb();
		};
	}
	if (existingId && cb) cb();
};

export const makeBinanceKycCall = (authToken: string) => {
	// @ts-ignore
	const binanceKyc = new BinanceKyc({
		authToken,
		bizEntityKey: BIZ_ENTRY_KEY,
		apiHost: BINANCE_DEV_URL,
		onMessage: ({ typeCode }: any) => {
			if (typeCode === '102') {
				binanceKyc.switchVisible(true);
			}
		}
	});
};

const getStorageValue = (key: string) => {
	const defaultValue = {
		access: '',
		refresh: '',
		account: '',
		isKyced: false
	} as LocalStorageProps;

	if (typeof window !== 'undefined') {
		const saved = localStorage.getItem(key);

		return saved ? (JSON.parse(saved) as LocalStorageProps) : defaultValue;
	}
};

export const useLocalStorage = (key: string) => {
	const [value, setValue] = useState(() => {
		return getStorageValue(key);
	});

	useEffect(() => {
		localStorage.setItem(key, JSON.stringify(value));
	}, [key, value]);

	return [value, setValue];
};

export const getAuthTokensFromNonce = async (account: string, library: any) => {
	try {
		const res = await axios.request({
			url: `${BASE_URL}${apiCall.getNonce}${account}`
		});
		try {
			const msg = getMetamaskMessage(res.data.nonce);
			const signature = await library?.send('personal_sign', [account, msg]);
			try {
				const tokenRes = await axios.request({
					url: `${BASE_URL}${apiCall.auth}`,
					method: 'POST',
					data: { address: account, signature }
				});

				return tokenRes.data as { access: string; is_kyced: boolean; refresh: string }; // TODO: define Type
			} catch (err: any) {
				throw new Error(err);
			}
		} catch (err: any) {
			throw new Error(err);
		}
	} catch (err: any) {
		throw new Error(err);
	}
};

export const useKyc = (
	authToken: string
): { loading: boolean; error: any; kycStatus: string; kycToken: string } => {
	const [loading, setLoading] = useState(false);
	const [error, setError] = useState(null);
	const [kycStatus, setKycStatus] = useState('');
	const [kycToken, setKycToken] = useState('');

	const fetchData = useCallback(
		async (authToken: string): Promise<void> => {
			try {
				setLoading(true);
				const statusRes = await axios.request({
					// TODO: check typing and if kycStatus is from right place
					url: `${BASE_URL}${apiCall.kycStatus}`,
					headers: {
						Authorization: `Bearer ${authToken}`
					}
				});
				setKycStatus(statusRes.data.statusInfo.kycStatus);
			} catch (err: any) {
				setError(err);
			} finally {
				setLoading(false);
			}

			try {
				setLoading(true);
				const tokenRes = await axios.request({
					url: `${BASE_URL}${apiCall.kycToken}`,
					headers: {
						Authorization: `Bearer ${authToken}`
					}
				});
				setKycToken(tokenRes.data.token);
			} catch (err: any) {
				setError(err);
			} finally {
				setLoading(false);
			}
		},
		[authToken]
	);

	useEffect(() => {
		void fetchData(authToken);
	}, [fetchData]);

	return { loading, error, kycStatus, kycToken };
};

export const useBinanceApi = () => {
	const [allPrices, setAllPrices] = useState<{ symbol: string; price: string }[]>([]);
	const [allFilteredPrices, setAllFilteredPrices] = useState<{ symbol: string; price: string }[]>(
		[]
	);
	const [allPairs, setAllPairs] = useState([]);
	const [allFilteredPairs, setAllFilteredPairs] = useState([]);

	const getExchangeInfo = async () => {
		try {
			const res = await axios.request({ url: BINANCE_EXCHANGE_INFO });
			setAllPairs(res.data.symbols);
		} catch (e: any) {
			throw new Error(e);
		}
	};

	const getTickerData = async () => {
		try {
			const res = await axios.request({ url: BINANCE_PRICE_TICKER });
			setAllPrices(res.data);
		} catch (e: any) {
			throw new Error(e);
		}
	};

	const uniqueTokens: string[] = Object.keys(destinationNetworks).reduce(
		(tokens: string[], network: string) => {
			// @ts-ignore
			const networkTokens = Object.keys(destinationNetworks?.[network]?.['tokens']);

			const allTokens = [...tokens, networkTokens];

			return [...new Set(allTokens.flat(1))];
		},
		['GLMR']
	);

	const isSymbol = (symbol: string): boolean => {
		let k = 0;
		for (let i = 0; i < uniqueTokens.length - 1; i++) {
			k++;
			for (let j = k; j < uniqueTokens.length; j++) {
				if (
					uniqueTokens[i] + uniqueTokens[j] === symbol ||
					uniqueTokens[j] + uniqueTokens[i] === symbol
				) {
					return true;
				}
			}
		}

		return false;
	};

	useEffect(() => {
		void getExchangeInfo();
		void getTickerData();
	}, []);

	useEffect(() => {
		if (allPairs) {
			const filteredPairs = allPairs.filter((pair: any) => isSymbol(pair.symbol));
			setAllFilteredPairs(filteredPairs);
		}

		if (allPrices) {
			const filteredPrices = allPrices.filter((price: any) => isSymbol(price.symbol));
			setAllFilteredPrices(filteredPrices);
		}
	}, [allPairs, allPrices]);

	return { allFilteredPairs, allFilteredPrices };
};
