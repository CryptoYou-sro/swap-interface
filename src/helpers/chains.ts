import { Chain } from '@wagmi/core';
 
export const moonbeam = {
    id: 1284,
    name: 'Moonbeam',
    network: 'moonbeam',
    nativeCurrency: {
        decimals: 18,
        name: 'GLMR',
        symbol: 'GLMR',
    },
    rpcUrls: {
        public: {
            http: ['https://rpc.api.moonbeam.network'],
            webSocket: ['wss://wss.api.moonbeam.network'],
        },
        default: {
            http: ['https://rpc.api.moonbeam.network'],
            webSocket: ['wss://wss.api.moonbeam.network'],
        },
    },
    blockExplorers: {
        default: {
            name: 'Moonscan',
            url: 'https://moonscan.io',
        },
        etherscan: {
            name: 'Moonscan',
            url: 'https://moonscan.io',
        },
    },
    contracts: {
        multicall3: {
            address: '0xcA11bde05977b3631167028862bE2a173976CA11',
            blockCreated: 609002,
        },
    },
    testnet: false
} as const satisfies Chain;

export const bsc = {
    id: 56,
    name: 'Binance Smart Chain',
    network: 'bsc',
    nativeCurrency: {
        decimals: 18,
        name: 'BNB',
        symbol: 'BNB',
    },
    rpcUrls: {
        public: {
            http: ['https://bsc.publicnode.com'],
        },
        default: {
            http: ['https://bsc.publicnode.com'],
        },
    },
    blockExplorers: {
        default: {
            name: 'BscScan',
            url: 'https://bscscan.com',
        },
        etherscan: {
            name: 'BscScan',
            url: 'https://bscscan.com',
        },
    },
    contracts: {
        multicall3: {
            address: '0xca11bde05977b3631167028862be2a173976ca11',
            blockCreated: 15921452,
        },
    },
    testnet: false
} as const satisfies Chain;
