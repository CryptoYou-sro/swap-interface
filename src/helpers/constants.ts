export const LOCAL_STORAGE_THEME = 'darkMode';
export const LOCAL_STORAGE_AUTH = 'auth';

export const INITIAL_STORAGE = {
	access: '',
	refresh: '',
	account: '',
	isKyced: false
};

export const MESSAGE = {
	ignore: 'ignore',
	failed: 'Request failed'
};

export const BIZ_ENTRY_KEY = 'BMYBIT';
export const BINANCE_PROD_URL = 'https://api.commonservice.io';
export const BINANCE_DEV_URL = 'https://dip-qacb.sdtaop.com';
export const BINANCE_SCRIPT =
	'https://static.saasexch.com/static/binance/static/kyc-ui/sdk/0.0.2/sdk.js';
export const BINANCE_PRICE_TICKER = 'https://www.binance.com/api/v3/ticker/price';
export const BINANCE_EXCHANGE_INFO = 'https://api.binance.com/api/v3/exchangeInfo';

export const BASE_URL = 'https://auth-app-aq3rv.ondigitalocean.app/';
// https://auth-app-aq3rv.ondigitalocean.app/account/withdraw/details?id=7ef91f76c79b4793b417e32fbffd73b8
export const MOONBEAM_URL = 'https://rpc.api.moonbeam.network';

export const routes = {
	getNonce: 'nonce?address=',
	auth: 'auth',
	kycToken: 'kyc/token',
	kycStatus: 'kyc/status',
	refresh: 'auth/refresh',
	transactionDetails: 'account/withdraw/details?id='
};

export const PROTOCOL_FEE = 0.002;
export const BINANCE_FEE = 0.002;
export const PROTOCOL_FEE_FACTOR = 1 / (1 - PROTOCOL_FEE);
export const START_TOKEN = 'GLMR';
export const FEE_CURRENCY = 'USDT';

export const ESTIMATED_NETWORK_TRANSACTION_GAS = 55_437;
export const SERVICE_ADDRESS = '0xeB56c1d19855cc0346f437028e6ad09C80128e02';

export const BLOCK_CONTRACT_NUMBER = 2_013_557;
export const WEI_TO_GLMR = 1 / 1000000000000000000;

export const CONTRACT_ADDRESSES = {
	1284: '0x6136F6c8FdC1c711892d217d0D5E95928dce6b20',
	1287: '0x2d249342F1F6D549CE8253782c8A2b9218c68fB2',
	31337: '0x0165878A594ca255338adfa4d48449f69242Eb8F'
};

export const BLOCKS_AMOUNT = 30;

export const makeId = (length: number) => {
	let result = '';
	const characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
	const charactersLength = characters.length;
	for (let i = 0; i < length; i++) {
		result += characters.charAt(Math.floor(Math.random() * charactersLength));
	}

	return result;
};
