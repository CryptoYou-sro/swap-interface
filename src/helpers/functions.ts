import type { Breakpoint, BreakpointOrNumber, Theme } from './../styles';
import { breakpoint } from './../styles';
import { useLayoutEffect, useState } from 'react';
import { format, utcToZonedTime } from 'date-fns-tz';

export const isLightTheme = (theme: Theme): boolean => theme.name === 'light';
const { timeZone } = Intl.DateTimeFormat().resolvedOptions();

export const isNetworkSelected = (network: string) =>
	network !== 'Select Network' && network !== ''; // TODO: refine both functions - nullish check - and create enum for "Select Network / Token"

export const isArrayType = (value: any) => typeof value === 'object' && Array.isArray(value);

export const isTokenSelected = (token: string) => token !== 'Select Token' && token !== '';

type BeautifyNumbers = {
	n: string | number;
	digits?: number;
};

const trimZeros = (res: string): string =>
	res.slice(-1) === '0' && res.slice(-2, -1) !== '.' ? trimZeros(res.slice(0, -1)) : res;

export const beautifyNumbers = ({ n, digits = 8 }: BeautifyNumbers): string => {
	let res = '';
	if (!n) return '';
	if (typeof n === 'number') {
		res = n.toFixed(digits);
	} else {
		res = (+n).toFixed(digits);
	}

	return trimZeros(res);
};

export const useWindowSize = () => {
	const [size, setSize] = useState([0, 0]);
	useLayoutEffect(() => {
		const updateSize = () => {
			setSize([window.innerWidth, window.innerHeight]);
		};
		window.addEventListener('resize', updateSize);
		updateSize();

		return () => window.removeEventListener('resize', updateSize);
	}, []);

	return size;
};

export const useBreakpoint = (size: BreakpointOrNumber) => {
	const [windowWidth, windowHeight] = useWindowSize();
	const isString = typeof size === typeof 'string';

	return {
		isBreakpointWidth: windowWidth < (isString ? breakpoint[size as Breakpoint] : size),
		isBreakpointHeight: windowHeight < (isString ? breakpoint[size as Breakpoint] : size)
	};
};

export const hexToRgbA = (hex: string, alpha = '1'): string => {
	const [r, g, b] = hex.match(/\w\w/g)!.map((x) => parseInt(x, 16));

	return `rgba(${r},${g},${b},${alpha})`;
};

export const isSwapRejected = (status: string, errorMessage: any) =>
	status === 'Exception' && errorMessage === 'user rejected transaction';

export const formatDate = (ts: number | undefined): string =>
	ts ? format(utcToZonedTime(new Date(ts * 1000), timeZone), 'dd/MM/yyyy HH:mm:ss') : 'n/a';
