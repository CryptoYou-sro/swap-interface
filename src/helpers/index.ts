export * from './functions';
export * from './context';
export * from './constants';
export * from './auth';
export * from './graph';
export * from './storage';
export * from './types';
