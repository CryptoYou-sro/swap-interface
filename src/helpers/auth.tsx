import { Buffer } from 'buffer';
import { BASE_URL, BINANCE_PROD_URL, BINANCE_SCRIPT, BIZ_ENTRY_KEY } from '../helpers';
import axios from 'axios';
import type { ApiAuthType } from '../helpers';

export enum STATUS_ENUM {
	NONCE = 'NONCE',
	AUTH = 'AUTH',
	PASS = 'PASS'
}

export const routes = {
	getNonce: 'nonce?address=',
	auth: 'auth',
	kycToken: 'kyc/token',
	kycStatus: 'kyc/status',
	refresh: 'auth/refresh'
};

export const getMetamaskMessage = (nonce: string): string =>
	`0x${Buffer.from('Please sign this one time nonce: ' + nonce, 'utf8').toString('hex')}`;

export const loadBinanceKycScript = (cb?: any) => {
	const existingId = document.getElementById('binance-kcy-script');

	if (!existingId) {
		const binanceSdkScript = document.createElement('script');
		binanceSdkScript.src = BINANCE_SCRIPT;
		binanceSdkScript.id = 'binance-kcy-script';
		document.body.appendChild(binanceSdkScript);

		binanceSdkScript.onload = () => {
			if (cb) cb();
		};
	}
	if (existingId && cb) cb();
};

export const makeBinanceKycCall = (authToken: string) => {
	// @ts-ignore
	const binanceKyc = new BinanceKyc({
		authToken,
		bizEntityKey: BIZ_ENTRY_KEY,
		apiHost: BINANCE_PROD_URL,
		onMessage: ({ typeCode }: any) => {
			if (typeCode === '102') {
				binanceKyc.switchVisible(true);
			}
		}
	});
};

export const getAuthTokensFromNonce = async (account: string, library: any) => {
	try {
		const res = await axios.request({
			url: `${BASE_URL}${routes.getNonce}${account}`
		});
		try {
			const msg = getMetamaskMessage(res.data.nonce);
			const signature = await library?.send('personal_sign', [account, msg]);
			try {
				const tokenRes = await axios.request({
					url: `${BASE_URL}${routes.auth}`,
					method: 'POST',
					data: { address: account, signature }
				});

				return tokenRes.data as ApiAuthType;
			} catch (err: any) {
				throw new Error(err);
			}
		} catch (err: any) {
			throw new Error(err);
		}
	} catch (err: any) {
		throw new Error(err);
	}
};
