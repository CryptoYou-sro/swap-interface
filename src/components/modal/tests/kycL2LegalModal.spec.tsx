import { fireEvent, render, waitFor } from '@testing-library/react';
import { publicProvider } from '@wagmi/core/dist/providers/public';
import { w3mConnectors } from '@web3modal/ethereum';

import {
    WagmiConfig,
    configureChains,
    createClient
} from 'wagmi';
import { mainnet } from 'wagmi/dist/chains';
import { AuthProvider } from '../../../helpers';
import { bsc, moonbeam } from '../../../helpers/chains';
import { KycL2LegalModal } from '../kycL2LegalModal';


describe('KycL2LegalModal', () => {
    const chains = [mainnet, moonbeam, bsc];
    const projectId = process.env.REACT_APP_WALLET_CONNECT_PROJECT_ID as string;
    const { provider } = configureChains(
        chains,
        [publicProvider()],
        { pollingInterval: 10_000 },
    );
    const wagmiClient = createClient({
        autoConnect: true,
        connectors: w3mConnectors({
            projectId,
            version: 1,
            chains
        }),
        provider
    });

    it('should render a kycL2 form for LEGAL persons', async () => {
        const {
            getByTestId,
            getByLabelText,
            getAllByLabelText,
            getByText,
            getAllByPlaceholderText,
            getAllByRole,
        } = render(
            <WagmiConfig client={wagmiClient}>
                <AuthProvider>
                    <KycL2LegalModal showKycL2={true} updateShowKycL2={() => console.log('modal')} />
                </AuthProvider>
            </WagmiConfig>
        );

        // PAGE 0 ======================================================================
        const nextBtn = getByText('Next');
        expect(nextBtn).toBeInTheDocument();
        expect(nextBtn).toBeDisabled();

        const companyNameInput = getByLabelText('Company name');
        expect(companyNameInput).toBeInTheDocument();
        fireEvent.change(companyNameInput, { target: { value: 'CryptoYou' } });
        // @ts-ignore
        expect(companyNameInput.value).toBe('CryptoYou');

        const companyIdNumberInput = getByLabelText('Company identification number');
        expect(companyIdNumberInput).toBeInTheDocument();
        fireEvent.change(companyIdNumberInput, { target: { value: '010101' } });
        // @ts-ignore
        expect(companyIdNumberInput.value).toBe('010101');

        const registeredOfficeTitle = 'Registered office';
        expect(getByText(registeredOfficeTitle)).toBeInTheDocument();

        const countryInput = getByLabelText('Country');
        expect(countryInput).toBeInTheDocument();
        fireEvent.change(countryInput, { target: { value: 'Albania' } });
        // @ts-ignore
        expect(countryInput.value).toBe('Albania');

        const streetInput = getByLabelText('Street');
        expect(streetInput).toBeInTheDocument();
        fireEvent.change(streetInput, { target: { value: 'Independence' } });
        // @ts-ignore
        expect(streetInput.value).toBe('Independence');

        const streetNumberInput = getByLabelText('Street number');
        expect(streetNumberInput).toBeInTheDocument();
        fireEvent.change(streetNumberInput, { target: { value: '01' } });
        // @ts-ignore
        expect(streetNumberInput.value).toBe('01');

        // FIXME: у модалці одночасно state і city 
        const stateInput = getByLabelText('State');
        expect(stateInput).toBeInTheDocument();
        fireEvent.change(stateInput, { target: { value: 'State custom' } });
        // @ts-ignore
        expect(stateInput.value).toBe('State custom');

        const cityInput = getByLabelText('City');
        expect(cityInput).toBeInTheDocument();
        fireEvent.change(cityInput, { target: { value: 'London' } });
        // @ts-ignore
        expect(cityInput.value).toBe('London');

        const zipcodeInput = getByLabelText('Zip Code');
        expect(zipcodeInput).toBeInTheDocument();
        fireEvent.change(zipcodeInput, { target: { value: 'zipcode custom' } });
        // @ts-ignore
        expect(zipcodeInput.value).toBe('zipcode custom');

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 1 ======================================================================
        const selectTaxResidencyInput = getByLabelText('Tax Residency');
        expect(selectTaxResidencyInput).toBeInTheDocument();
        const optionsTaxResidency = Array.from(selectTaxResidencyInput.querySelectorAll('option'));
        expect(optionsTaxResidency).toHaveLength(244);
        fireEvent.change(selectTaxResidencyInput, { target: { value: 'Albania' } });
        // @ts-ignore
        expect(selectTaxResidencyInput.value).toBe('Albania');

        const permanentRadioBtnTrue = getByLabelText('Yes');
        expect(permanentRadioBtnTrue).toBeInTheDocument();
        // @ts-ignore
        expect(permanentRadioBtnTrue.checked).toBe(true);

        const permanentRadioBtnFalse = getByLabelText('No');
        expect(permanentRadioBtnFalse).toBeInTheDocument();
        fireEvent.click(permanentRadioBtnFalse);
        await waitFor(() => {
            // @ts-ignore
            expect(permanentRadioBtnFalse.checked).toBe(true);
            // @ts-ignore
            expect(permanentRadioBtnTrue.checked).toBe(false);
        });

        const mailCountryInput = getByLabelText('Country');
        expect(mailCountryInput).toBeInTheDocument();
        fireEvent.change(mailCountryInput, { target: { value: 'American Samoa' } });
        // @ts-ignore
        expect(mailCountryInput.value).toBe('American Samoa');

        const mailStreetInput = getByLabelText('Street');
        expect(mailStreetInput).toBeInTheDocument();
        fireEvent.change(mailStreetInput, { target: { value: 'mail street custom' } });
        // @ts-ignore
        expect(mailStreetInput.value).toBe('mail street custom');

        const mailZipCodeInput = getByLabelText('ZIP Code');
        expect(mailZipCodeInput).toBeInTheDocument();
        fireEvent.change(mailZipCodeInput, { target: { value: 'mail zip code custom' } });
        // @ts-ignore
        expect(mailZipCodeInput.value).toBe('mail zip code custom');

        const mailCityInput = getByLabelText('City');
        expect(mailCityInput).toBeInTheDocument();
        fireEvent.change(mailCityInput, { target: { value: 'mail city London custom' } });
        // @ts-ignore
        expect(mailCityInput.value).toBe('mail city London custom');

        const mailStreetNumberInput = getByLabelText('Street number');
        expect(mailStreetNumberInput).toBeInTheDocument();
        fireEvent.change(mailStreetNumberInput, { target: { value: '02' } });
        // @ts-ignore
        expect(mailStreetNumberInput.value).toBe('02');

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 2 ======================================================================
        expect(nextBtn).toBeDisabled();

        const page2Title = getByText('Who is representing the company?');
        expect(page2Title).toBeInTheDocument();

        const representingCompanyInput = getByLabelText('Statutory body');
        expect(representingCompanyInput).toBeInTheDocument();

        fireEvent.click(representingCompanyInput);
        // @ts-ignore
        expect(representingCompanyInput.checked).toBe(true);

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 3 ======================================================================
        const areasTitle = getByText('Select in which areas your company is conducting activities');
        expect(areasTitle).toBeInTheDocument();

        const armsTransactionInput = getByLabelText('Arms Transaction');
        expect(armsTransactionInput).toBeInTheDocument();
        // @ts-ignore
        expect(armsTransactionInput.checked).toBe(false);

        fireEvent.click(armsTransactionInput);
        // @ts-ignore
        expect(armsTransactionInput.checked).toBe(true);

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 4 ======================================================================
        const countryInputs = getAllByRole('combobox');
        // NOTE: REACT SELECT IS UNSUPPORTING TESTS (custom library);
        countryInputs.forEach((countryInput) => {
            expect(countryInput).toBeInTheDocument();

            fireEvent.change(countryInput, { target: { value: 'Albania' } });
            // @ts-ignore
            expect(countryInput.value).toBe('Albania');
        });
        // @ts-ignore
        expect(countryInputs[0].value).toBe('Albania');
        // @ts-ignore
        expect(countryInputs[1].value).toBe('Albania');

        fireEvent.click(nextBtn);

        // PAGE 5 ======================================================================
        const sourceOfIncomeTitle = getByText('Source of income');
        expect(sourceOfIncomeTitle).toBeInTheDocument();

        const page5CheckBoxes = getAllByLabelText('Other');
        page5CheckBoxes.forEach((checkbox) => {
            // @ts-ignore
            expect(checkbox.checked).toBe(false);
            fireEvent.click(checkbox);
            // @ts-ignore
            expect(checkbox.checked).toBe(true);
        });

        const sourceOfIncomeSpecifyInputs = getAllByPlaceholderText('Specify...');

        sourceOfIncomeSpecifyInputs.forEach((input) => {
            // @ts-ignore
            fireEvent.change(input, { target: { value: 'It and development' } });
            // @ts-ignore
            expect(input.value).toBe('It and development');
        });

        const yearlyIncomeRadio = getByLabelText('Up to EUR 40,000');
        expect(yearlyIncomeRadio).toBeInTheDocument();
        fireEvent.click(yearlyIncomeRadio);
        await waitFor(() => {
            // @ts-ignore
            expect(yearlyIncomeRadio.checked).toBe(true);
        });
        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 6 ======================================================================
        expect(nextBtn).toBeDisabled();

        const yearlyIncomeRadioTrue = getByLabelText('Yes');
        const yearlyIncomeRadioFalse = getByLabelText('No');

        expect(yearlyIncomeRadioTrue).toBeInTheDocument();
        expect(yearlyIncomeRadioFalse).toBeInTheDocument();

        fireEvent.click(yearlyIncomeRadioFalse);
        await waitFor(() => {
            // @ts-ignore
            expect(yearlyIncomeRadioFalse.checked).toBe(true);
        });

        const naturalPersonRadioBtn = getByLabelText('Natural Person');
        const legalEntityRadioBtn = getByLabelText('Legal entity');

        expect(naturalPersonRadioBtn).toBeInTheDocument();
        expect(legalEntityRadioBtn).toBeInTheDocument();

        fireEvent.click(naturalPersonRadioBtn);
        await waitFor(() => {
            // @ts-ignore
            expect(naturalPersonRadioBtn.checked).toBe(true);
        });

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 7 ======================================================================
        const addUboBtn = getByText('Add UBO');
        expect(addUboBtn).toBeInTheDocument();

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);
        // PAGE 8 ======================================================================
        const addShareHolderBtn = getByText('Add shareholder');
        expect(addShareHolderBtn).toBeInTheDocument();

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);

        // PAGE 9 ======================================================================
        const addMemberBtn = getByText('Add member');
        expect(addMemberBtn).toBeInTheDocument();

        expect(nextBtn).not.toBeDisabled();
        fireEvent.click(nextBtn);


        // PAGE 10 ======================================================================

        const fileInputs = getAllByLabelText('Upload File', { selector: 'input[type="file"]' });

        fileInputs.forEach((input) => {
            fireEvent.change(input, { target: { files: [new File(['file content'], '123.txt', { type: 'text/plain' })] } });
        });

        const submitBtn = getByText('Submit');
        expect(submitBtn).toBeInTheDocument();
        expect(submitBtn).not.toBeDisabled();

        expect(getByTestId('kycL2LegalModalTest')).toMatchSnapshot();
    });
});