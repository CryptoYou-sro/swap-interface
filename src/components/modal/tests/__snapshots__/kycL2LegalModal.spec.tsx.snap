// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KycL2LegalModal should render a kycL2 form for LEGAL persons 1`] = `
<div
  class="sc-dycYrt kvpFnv"
  data-testid="kycL2LegalModalTest"
>
  <div
    style="display: flex; width: 100%; height: 100%; flex-direction: column; justify-content: space-between; align-items: center;"
  >
    <div
      class="sc-jOnpCo bZwdfQ"
      style="display: flex; flex-direction: column; width: 100%;"
    >
      <h2
        class="sc-hTUWRQ lipcwn"
      >
        Business verification
      </h2>
      <div
        style="margin: 0px 0px 10px; text-align: left; display: flex; align-items: baseline; flex-wrap: wrap;"
      >
        <p
          class="sc-lizKOf fvYQYa"
          style="max-width: 75%; margin-right: 10px;"
        >
          Copy of an account statement kept by an institution in the European Economic Area
        </p>
        <div
          style="text-align: left; margin-bottom: 40px; display: flex; align-items: stretch;"
        >
          <label
            class="sc-eoVZPG hmvdnO"
            for="file-input-refPoaDoc1"
          >
            <input
              class="sc-dWZqqJ jnLnFB"
              id="file-input-refPoaDoc1"
              type="file"
            />
            123.txt
          </label>
          <div
            class="sc-dhFUGM hGQIIk"
          >
            <div
              class="sc-jnOGJG iEiuXy"
              data-testid="icon"
              style="outline: none;"
            >
              <svg
                theme="[object Object]"
              >
                trash.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin: 0px 0px 10px; text-align: left; display: flex; align-items: baseline; flex-wrap: wrap;"
      >
        <p
          class="sc-lizKOf fvYQYa"
          style="max-width: 75%; margin-right: 10px;"
        >
          Documents proving information on the source of funds (i.e.: payslip, tax return etc.)
        </p>
        <div
          style="text-align: left; margin-bottom: 40px; display: flex; align-items: stretch;"
        >
          <label
            class="sc-eoVZPG hmvdnO"
            for="file-input-refPosofDoc1"
          >
            <input
              class="sc-dWZqqJ jnLnFB"
              id="file-input-refPosofDoc1"
              type="file"
            />
            123.txt
          </label>
          <div
            class="sc-dhFUGM hGQIIk"
          >
            <div
              class="sc-jnOGJG iEiuXy"
              data-testid="icon"
              style="outline: none;"
            >
              <svg
                theme="[object Object]"
              >
                trash.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin: 0px 0px 10px; text-align: left; display: flex; align-items: baseline; flex-wrap: wrap;"
      >
        <p
          class="sc-lizKOf fvYQYa"
          style="max-width: 75%; margin-right: 10px;"
        >
          Copy of excerpt of public register or other valid documents proving the existence of legal entity (Articles of Associations, Deed of Foundation etc.).
        </p>
        <div
          style="text-align: left; margin-bottom: 40px; display: flex; align-items: stretch;"
        >
          <label
            class="sc-eoVZPG hmvdnO"
            for="file-input-refPorDoc1"
          >
            <input
              class="sc-dWZqqJ jnLnFB"
              id="file-input-refPorDoc1"
              type="file"
            />
            123.txt
          </label>
          <div
            class="sc-dhFUGM hGQIIk"
          >
            <div
              class="sc-jnOGJG iEiuXy"
              data-testid="icon"
              style="outline: none;"
            >
              <svg
                theme="[object Object]"
              >
                trash.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        style="margin: 0px 0px 10px; text-align: left; display: flex; align-items: baseline; flex-wrap: wrap;"
      >
        <p
          class="sc-lizKOf fvYQYa"
          style="max-width: 75%; margin-right: 10px;"
        >
          Court decision on appointment of legal guardian (if relevant).
        </p>
        <div
          style="text-align: left; margin-bottom: 40px; display: flex; align-items: stretch;"
        >
          <label
            class="sc-eoVZPG hmvdnO"
            for="file-input-refPogDoc1"
          >
            <input
              class="sc-dWZqqJ jnLnFB"
              id="file-input-refPogDoc1"
              type="file"
            />
            123.txt
          </label>
          <div
            class="sc-dhFUGM hGQIIk"
          >
            <div
              class="sc-jnOGJG iEiuXy"
              data-testid="icon"
              style="outline: none;"
            >
              <svg
                theme="[object Object]"
              >
                trash.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="margin: 0px auto; width: 100%; text-align: center;"
    >
      <button
        class="sc-bkEOxz gPpopC"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
